device_backends/include/DummyBackendBase.h

```diff
@@ -103,17 +104,51 @@ namespace ChimeraTK {
     template<typename UserType>
     boost::shared_ptr<NDRegisterAccessor<UserType>> getRegisterAccessor_impl(const RegisterPath& registerPathName,
         size_t numberOfWords, size_t wordOffsetInRegister, AccessModeFlags flags) {
+
+      // First check if the request is for one of the special DUMMY_INTEERRUPT_X_Y registers. if so, early return
+      // this special accessor.
+      // Pseudo-register to trigger interrupts for this register
+      static const std::string DUMMY_INTERRUPT_REGISTER_NAME{"^/DUMMY_INTERRUPT_([0-9]+)_([0-9]+)$"};
+
+      const std::string regPathNameStr{registerPathName};
+      const std::regex dummyInterruptRegex{DUMMY_INTERRUPT_REGISTER_NAME};
+      std::smatch match;
+      std::regex_search(regPathNameStr, match, dummyInterruptRegex);
+
+      if(not match.empty()) {
+        // FIXME: Ideally, this test and the need for passing in the lambda function should be done
+        // in the constructor of the accessor. But passing down the base-class of the accessor is very weird
+        // due to the sort-of CRTP pattern used in this base class.
+        auto controller = std::stoi(match[1].str());
+        auto interrupt = std::stoi(match[2].str());
+        try {
+          auto& interruptsForController = _registerMap->getListOfInterrupts().at(controller);
+          if(interruptsForController.find(interrupt) == interruptsForController.end())
+            throw std::out_of_range("Invalid interrupt for controller");
+        }
+        catch(std::out_of_range&) {
```

Comment:
throwing std::out_of_range and catching it again looks weird. I understand that you do it because std::map::at() can also throw. A comment would be appropriate